components/suggest/src/rs.rs

```diff
@@ -69,85 +72,58 @@ pub(crate) trait Client {
-    quicksuggest_client: remote_settings::RemoteSettings,
-    fakespot_client: remote_settings::RemoteSettings,
+    quicksuggest_client: Arc<RemoteSettingsClient>,
+    fakespot_client: Arc<RemoteSettingsClient>,
 }
 
-impl RemoteSettingsClient {
-    pub fn new(
-        server: Option<remote_settings::RemoteSettingsServer>,
-        bucket_name: Option<String>,
-        server_url: Option<String>,
-    ) -> Result<Self> {
+impl SuggestRemoteSettingsClient {
+    pub fn new(rs_service: &RemoteSettingsService) -> Result<Self> {
         Ok(Self {
-            quicksuggest_client: remote_settings::RemoteSettings::new(
-                remote_settings::RemoteSettingsConfig {
-                    server: server.clone(),
-                    bucket_name: bucket_name.clone(),
-                    collection_name: "quicksuggest".to_owned(),
-                    server_url: server_url.clone(),
-                },
-            )?,
-            fakespot_client: remote_settings::RemoteSettings::new(
-                remote_settings::RemoteSettingsConfig {
-                    server,
-                    bucket_name,
-                    collection_name: "fakespot-suggest-products".to_owned(),
-                    server_url,
-                },
-            )?,
+            quicksuggest_client: rs_service.make_client("quicksuggest".to_owned())?,
+            fakespot_client: rs_service.make_client("fakespot-suggest-products".to_owned())?,
         })
     }
 
-    fn client_for_collection(&self, collection: Collection) -> &remote_settings::RemoteSettings {
+    fn client_for_collection(&self, collection: Collection) -> &RemoteSettingsClient {
         match collection {
             Collection::Fakespot => &self.fakespot_client,
             Collection::Quicksuggest => &self.quicksuggest_client,
         }
     }
 }
 
-impl Client for RemoteSettingsClient {
-    fn get_records(&self, collection: Collection, dao: &mut SuggestDao) -> Result<Vec<Record>> {
+impl Client for SuggestRemoteSettingsClient {
+    fn get_records(&self, collection: Collection) -> Result<Vec<Record>> {
         // For now, handle the cache manually.  Once 6328 is merged, we should be able to delegate
         // this to remote_settings.
         let client = self.client_for_collection(collection);
-        let cache = dao.read_cached_rs_data(collection.name());
-        let last_modified = match &cache {
-            Some(response) => response.last_modified,
-            None => 0,
-        };
-        let response = match cache {
-            None => client.get_records()?,
-            Some(cache) => remote_settings::cache::merge_cache_and_response(
-                cache,
-                client.get_records_since(last_modified)?,
-            ),
-        };
-        if last_modified != response.last_modified {
-            dao.write_cached_rs_data(collection.name(), &response);
+        client.sync()?;
+        let response = client.get_records(false);
```

Comment:
Just for my own information -- does this new RS client periodically sync on its own? (No, right?) Say we didn't call `sync()` here and instead passed true for `sync_if_empty`. That would mean we would do an initial sync but AFAICT nothing would ever be synced after that, right?

Reply:
Yes that's right.  I think the system we should be eventually moving towards is the application calls `RemoteSettingsService::sync` on some periodic schedule.  However, that's not happening yet so clients have to call `sync()` themselves.